libraries/AP_Notify/Display.cpp

```diff
@@ -385,65 +438,97 @@ void Display::draw_char(uint16_t x, uint16_t y, const char c)
         }
     }
 
-    // update display
-    _need_update = true;
+
 }
 
-void Display::update_arm()
+void Display::update_arm(uint8_t r)
 {
     if (AP_Notify::flags.armed) {
-        draw_text(COLUMN(0), ROW(0), ">>>>> ARMED! <<<<<");
+        draw_text(COLUMN(0), ROW(r), ">>>>> ARMED! <<<<<");
     } else {
-        draw_text(COLUMN(0), ROW(0), "     disarmed     ");
+        draw_text(COLUMN(0), ROW(r), "     disarmed     ");
     }
 }
 
-void Display::update_prearm()
+void Display::update_prearm(uint8_t r)
 {
     if (AP_Notify::flags.pre_arm_check) {
-        draw_text(COLUMN(0), ROW(2), "Prearm: passed    ");
+        draw_text(COLUMN(0), ROW(r), "Prearm: passed    ");
     } else {
-        draw_text(COLUMN(0), ROW(2), "Prearm: failed    ");
+        draw_text(COLUMN(0), ROW(r), "Prearm: failed    ");
     }
 }
 
-void Display::update_gps()
+void Display::update_gps(uint8_t r)
 {
-    switch (AP_Notify::flags.gps_status) {
-    case AP_GPS::NO_GPS:
-        draw_text(COLUMN(0), ROW(3), "GPS:    no GPS");
-        break;
-    case AP_GPS::NO_FIX:
-        draw_text(COLUMN(0), ROW(3), "GPS:    no fix");
-        break;
-    case AP_GPS::GPS_OK_FIX_2D:
-        draw_text(COLUMN(0), ROW(3), "GPS:    2D    ");
-        break;
-    case AP_GPS::GPS_OK_FIX_3D:
-        draw_text(COLUMN(0), ROW(3), "GPS:    3D    ");
-        break;
-    case AP_GPS::GPS_OK_FIX_3D_DGPS:
-        draw_text(COLUMN(0), ROW(3), "GPS:    DGPS  ");
-        break;
-    case AP_GPS::GPS_OK_FIX_3D_RTK:
-        draw_text(COLUMN(0), ROW(3), "GPS:    RTK   ");
-        break;
-    }
+	static const char * gpsfixname[] = {"NoGPS","NoFix","2D   ","3D   ","DGPS " ,"RTK "};
+	char msg [DISPLAY_MESSAGE_SIZE];
+	snprintf(msg, DISPLAY_MESSAGE_SIZE, "GPS:%s Sats:%u", gpsfixname[AP_Notify::flags.gps_status], AP_Notify::flags.gps_num_sats) ;
```

Comment:
This is the enum: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_GPS/AP_GPS.h#L76 It will be changing in the next week or so, gaining a bunch of additional modes.